src/util/FeedSwitch.sol

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.18;
+
+import "src/interfaces/IChainlinkFeed.sol";
+
+/// @title FeedSwitch
+/// @notice A contract to switch between feeds after a timelock period
+/// @dev The contract can only be initiated by the guardian and the feed can only be switched after the timelock period has passed.
+/// The backup feed will default to the afterMaturityFeed if the maturity has passed, if not it will default to the beforeMaturityFeed.
+/// If a switch is done before maturity, the guardian can initiate the switch again to update the feed to the afterMaturityFeed if the maturity has passed.
+contract FeedSwitch {
+    error SwitchNotInitiated();
+    error CannotSwitchYet();
+    error NotGuardian();
+    error FeedDecimalsMismatch();
+
+    address public immutable guardian;
+    IChainlinkFeed public immutable beforeMaturityFeed;
+    IChainlinkFeed public immutable afterMaturityFeed;
+    uint256 public immutable maturity;
+    uint256 public immutable timelockPeriod;
+
+    IChainlinkFeed public feed;
+    uint256 public switchInitiatedAt;
+
+    event FeedSwitchInitiated();
+    event FeedSwitched(address indexed newFeed);
+
+    constructor(
+        address _feed,
+        address _beforeMaturityFeed,
+        address _afterMaturityFeed,
+        uint256 _timelockPeriod,
+        uint256 _maturity,
+        address _guardian
+    ) {
+        feed = IChainlinkFeed(_feed);
+        beforeMaturityFeed = IChainlinkFeed(_beforeMaturityFeed);
+        afterMaturityFeed = IChainlinkFeed(_afterMaturityFeed);
+        if (
+            feed.decimals() != beforeMaturityFeed.decimals() ||
+            feed.decimals() != afterMaturityFeed.decimals() ||
+            feed.decimals() != 18
+        ) revert FeedDecimalsMismatch();
+
+        timelockPeriod = _timelockPeriod;
+        maturity = _maturity;
```

Comment:
Not super important, but consider a constructor check that makes sure that maturity is in the future.